From 678c81271b266b955211791e70fe5f16feacb0e0 Mon Sep 17 00:00:00 2001
From: Sean McBride <sean@rogue-research.com>
Date: Sat, 3 Feb 2024 22:36:12 -0500
Subject: [PATCH] descriptor: Small clarifications with no behaviour change

All the right hand side is `dev_cap`, change one outlier to match.

Also clarify the relationships between some magic numbers.

No change in behaviour here.
---
 libusb/descriptor.c   | 4 ++--
 libusb/version_nano.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libusb/descriptor.c b/libusb/descriptor.c
index d3aa9398a..5b83cc982 100644
--- a/libusb/descriptor.c
+++ b/libusb/descriptor.c
@@ -1208,11 +1208,11 @@ int API_EXPORTED libusb_get_platform_descriptor(libusb_context *ctx,
 	_platform_descriptor->bReserved = dev_cap->dev_capability_data[0];
 	memcpy(_platform_descriptor->PlatformCapabilityUUID, &(dev_cap->dev_capability_data[1]), 16);
 
-	/* Capability data is located after reserved byte and 128-bit UUID */
+	/* Capability data is located after reserved byte and 16 byte UUID */
 	uint8_t* capability_data = dev_cap->dev_capability_data + 1 + 16;
 
 	/* Capability data length is total descriptor length minus initial fields */
-	size_t capability_data_length = _platform_descriptor->bLength - (16 + 4);
+	size_t capability_data_length = dev_cap->bLength - (3 + 1 + 16);
 
 	memcpy(_platform_descriptor->CapabilityData, capability_data, capability_data_length);
 
diff --git a/libusb/version_nano.h b/libusb/version_nano.h
index 2dc4c93f2..6fee87bbe 100644
--- a/libusb/version_nano.h
+++ b/libusb/version_nano.h
@@ -1 +1 @@
-#define LIBUSB_NANO 11910
+#define LIBUSB_NANO 11911
